Flatpak updates

worldofpeace opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
See commit messages and release notes.
Lots of changes, new portals etc. (had to regenerate every patch)

With some light testing I think the background portal is causing some issues
in the gnome3 session, so this is a draft until I can fix that issue.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace committed:
flatpak: 1.2.4 -> 1.4.2

* Regenerated all patches for 1.4.2 and resolved
  any conflicts.

* fix-test-paths.patch doesn't copy the whole locale archive
  because we have C.UTF8 now.

* nixos/flatpak creates a Flatpak system helper user
  Change introduced in 1.3.2.

Changes:
See https://github.com/flatpak/flatpak/releases/tag/1.3.1 through
1.4.2.

worldofpeace committed:
xdg-desktop-portal: 1.2.0 -> 1.4.2

* adapt respect-path-env-var.patch to code being
  moved around, etc.

* Add  json-glib

https://github.com/flatpak/xdg-desktop-portal/releases/tag/1.4.0

worldofpeace committed:
xdg-desktop-portal-gtk: 1.2.0 -> 1.4.0

Stylize expression.

https://github.com/flatpak/xdg-desktop-portal-gtk/releases/tag/1.4.0

jtojnar reviewed (commented):
From https://github.com/flatpak/flatpak/releases/tag/1.3.4:

> Improve ability to pull from multiple p2p sources (needs latest ostree).

Maybe we need to update that as well.

worldofpeace commented on pkgs/development/libraries/flatpak/fix-test-paths.patch:
```diff
@@ -105,21 +111,20 @@
-+# We copy the whole locale archive because we do not have C.UTF8 locale
- mkdir -p ${DIR}/usr/lib/locale/
--cp -r /usr/lib/locale/C.* ${DIR}/usr/lib/locale/en_US
-+cp @glibcLocales@/lib/locale/locale-archive ${DIR}/usr/lib/locale/locale-archive
```
Perhaps I can trying running the test to see.

jtojnar replied:
Most Flatpak tests seem to pass so perhaps `en_US` is not really needed.

worldofpeace commented on nixos/modules/services/desktops/flatpak.nix:
```diff
@@ -32,5 +32,13 @@ in {
       "/var/lib/flatpak/exports"
     ];
+
+    users.users.flatpak = {
```
Yep, will add a comment about SELinux.

worldofpeace commented:
> Maybe we need to update that (ostree) as well.

Totally forgot about that one. Will do.

worldofpeace committed:
ostree: 2019.1 -> 2019.2

All patches synced.
fix-1592.patch needed some new additions since that
issue affects more test cases.

https://github.com/ostreedev/ostree/releases/tag/v2019.2

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

worldofpeace committed:
libmodulemd: init at 2.6.0

Includes a patch for the usual pygobject override stuff.
Pending upstream submission.

worldofpeace committed:
libmodulemd_1: init at 1.8.15

worldofpeace committed:
rpm-ostree: 2019.2 -> 2019.5

https://github.com/projectatomic/rpm-ostree/releases/tag/v2019.3
https://github.com/projectatomic/rpm-ostree/releases/tag/v2019.4
https://github.com/projectatomic/rpm-ostree/releases/tag/v2019.5

worldofpeace commented:
7f8e106834be69161a4032b430e6d7ebbb4d9586...dba2ca26579635152a5e9d7c5789efa0777dbc01 has more patches than I'd like.

Which is a problem I noticed with `ostree`, and I didn't really bother to fix the tests there.

worldofpeace committed:
nixos/flatpak: add comment about selinux

worldofpeace commented on pkgs/development/libraries/libmodulemd/dont-check-docs.patch:
```diff
@@ -0,0 +1,29 @@
+-if ret.returncode() != 0
+-  error('Missing documentation for GObject.')
+-endif
++# ret = run_command ([test, '-e', join_paths(glib_docpath, 'glib/index.html')])
```
Commented because these checks are kinda extra

jtojnar replied:
This is a common pain point when I use DevHelp. We need to come up with a way to fill in the paths and disseminate it upstream.

worldofpeace commented:
Some of the errors I see in a vm

<details><summary>errors</summary>

##### These are in the x11 session.

```
Failed to get application states: GDBus.Error:org.freedesktop.portal.Error.Failed: Could not get window list: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: App introspection not allowed
```

```
 .xdg-desktop-po[976]: No skeleton to export
```

```
.xdg-desktop-po[976]: Failed to create file chooser proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gtk: Timeout was reached
```

##### These are in the wayland session
For some reason it stalls to start for a very long time.
Perhaps it's because a service timesout to start.

```
 .gsd-wacom-wrap[1214]: Settings portal not found: Error calling StartServiceByName for org.freedesktop.portal.Desktop: Timeout was reached
```

```
 .gnome-shell-wr[1079]: AT-SPI: Could not obtain desktop path or name
```

```
 .evolution-sour[1113]: Unable to connect to the GNOME Online Accounts service: Error calling StartServiceByName for org.gnome.OnlineAccounts: Timeout was reached
```

Probably a prime suspect
```
systemd[715]: xdg-permission-store.service: Main process exited, code=killed, status=15/TERM
```

```
systemd-logind[622]: Session c1 logged out. Waiting for processes to exit.
at-spi-bus-launcher[839]: XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":1024"
at-spi-bus-launcher[839]:       after 21 requests (21 known processed) with 0 events remaining.
.gsd-clipboard-[879]: gsd-clipboard: Fatal IO error 11 (Resource temporarily unavailable) on X server :1024.
 .gsd-media-keys[897]: gsd-media-keys: Fatal IO error 11 (Resource temporarily unavailable) on X server :1024.
 .gsd-color-wrap[886]: gsd-color: Fatal IO error 11 (Resource temporarily unavailable) on X server :1024.
.gsd-power-wrap[883]: gsd-power: Fatal IO error 11 (Resource temporarily unavailable) on X server :1024.
.gsd-keyboard-w[889]: gsd-keyboard: Fatal IO error 11 (Resource temporarily unavailable) on X server :1024.
.gsd-xsettings-[880]: gsd-xsettings: Fatal IO error 11 (Resource temporarily unavailable) on X server :1024.
.gnome-shell-wr[826]: Connection to xwayland lost
polkitd[671]: Unregistered Authentication Agent for unix-session:c1 (system bus name :1.31, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UT>
systemd[1]: session-c1.scope: Succeeded.
```

Too many more, what a headache :smile: 

</details>

jtojnar commented:
After re-adding `fix-1592.patch` the ostree tests only fail with `FAIL: libostree/test-config.sh.test`.

Running `nix run -f . gnome-desktop-testing ostree -c gnome-desktop-testing-runner -d $(nix-build -A ostree.installedTests --no-out-link)/share libostree/test-config.sh.test` pointed me out that the https://github.com/ostreedev/ostree/issues/1592 crept out to more test cases, hopefully this fixes it.

worldofpeace commented:
> After re-adding `fix-1592.patch` the ostree tests only fail with `FAIL: libostree/test-config.sh.test`.
> 
> Running `nix run -f . gnome-desktop-testing ostree -c gnome-desktop-testing-runner -d $(nix-build -A ostree.installedTests --no-out-link)/share libostree/test-config.sh.test` pointed me out that the [ostreedev/ostree#1592](https://github.com/ostreedev/ostree/issues/1592) crept out to more test cases, hopefully this fixes it.
> 
> **Edit:** Yeah, `nix-build '<nixpkgs/nixos/tests/ostree.nix>'` now passes.

Yeah, I got up to the point of 
> After re-adding fix-1592.patch the ostree tests only fail with FAIL: libostree/test-config.sh.test.

and gave up :smile: 

Good to see they're "fixed". Any ideas on how there could be a change upstream so we don't have to keep this patch?

jtojnar commented:
Hmm, weird. I can see the interface when I try to save a web page in Epiphany:

![Screenshot from 2019-07-31 14-45-40](https://user-images.githubusercontent.com/705123/62213014-4d08b180-b3a2-11e9-84f9-e72957248560.png)

But it still fails:

```
Can't open portal file chooser: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.FileChooser” on object at path /org/freedesktop/portal/desktop
```

Actually, I can reproduce this on my production system, is that possibly a regression from #64575?

worldofpeace commented:
Yeah, I should rebase this branch since that's all be worked out now.

worldofpeace commented:
I've confirmed that the issue I had with the wayland session and flatpak was a problem before this, and I believe I have another reporters confirmation of this https://github.com/NixOS/nixpkgs/pull/65449#issuecomment-517034146.

I will undraft this.

jtojnar commented on pkgs/tools/misc/rpm-ostree/default.nix:
```diff
@@ -28,6 +28,8 @@ stdenv.mkDerivation rec {
     json-glib libarchive libsolv librepo
     pcre check python
+     # libdnf
+    json_c libmodulemd_1 utillinux sqlite cppunit
```
Why is this required? I do not see `libmodulemd_1` to be used anywhere.

worldofpeace replied:
The comment above says `libdnf` because there's the subproject https://github.com/rpm-software-management/libdnf/tree/8a6b5837bb868ee398cd5bfc76a1d9668da0a050 that I believe depends on it.

jtojnar replied:
You are right, it is in the submodule which I did not have checked out.

jtojnar commented:
Now only `flatpak/test-extensions.sh.test` seems to be failing:


```
machine: must succeed: env FLATPAK_TESTS_DEBUG=1 gnome-desktop-testing-runner -d '/nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/share' --timeout 3600 flatpak/test-extensions.sh.test 2>&1
machine# [   21.018802] systemd[1]: NetworkManager-dispatcher.service: Succeeded.
machine# [   23.005946] gnome-desktop-testing-runner[874]: Executing: flatpak/test-extensions.sh.test
machine# [   28.006748] gnome-desktop-testing-runner[874]: Executing: flatpak/test-extensions.sh.test
machine# [   29.987314] gnome-desktop-testing-runner[874]: FAIL: flatpak/test-extensions.sh.test (Child process exited with code 1)
machine# [   30.016190] gnome-desktop-testing-runner[874]: SUMMARY: total=1; passed=0; skipped=0; failed=1; user=0.4s; system=1.9s; maxrss=16428
machine: exit status 2
machine: output: Running test: flatpak/test-extensions.sh.test
++ test -n ''
++ test -n ''
++ CMD_PREFIX=
++ export MALLOC_CHECK_=3
++ MALLOC_CHECK_=3
++ export MALLOC_PERTURB_=173
++ MALLOC_PERTURB_=173
+++ mktemp -d /tmp/test-flatpak-XXXXXX
++ TEST_DATA_DIR=/tmp/test-flatpak-JBDUYH
++ mkdir -p /tmp/test-flatpak-JBDUYH/home
++ mkdir -p /tmp/test-flatpak-JBDUYH/runtime
++ mkdir -p /tmp/test-flatpak-JBDUYH/system
++ mkdir -p /tmp/test-flatpak-JBDUYH/config
++ export FLATPAK_SYSTEM_DIR=/tmp/test-flatpak-JBDUYH/system
++ FLATPAK_SYSTEM_DIR=/tmp/test-flatpak-JBDUYH/system
++ export FLATPAK_SYSTEM_CACHE_DIR=/tmp/test-flatpak-JBDUYH/system-cache
++ FLATPAK_SYSTEM_CACHE_DIR=/tmp/test-flatpak-JBDUYH/system-cache
++ export FLATPAK_SYSTEM_HELPER_ON_SESSION=1
++ FLATPAK_SYSTEM_HELPER_ON_SESSION=1
++ export FLATPAK_CONFIG_DIR=/tmp/test-flatpak-JBDUYH/config
++ FLATPAK_CONFIG_DIR=/tmp/test-flatpak-JBDUYH/config
++ export FLATPAK_FANCY_OUTPUT=0
++ FLATPAK_FANCY_OUTPUT=0
++ export HOME=/tmp/test-flatpak-JBDUYH/home
++ HOME=/tmp/test-flatpak-JBDUYH/home
++ export XDG_CACHE_HOME=/tmp/test-flatpak-JBDUYH/home/cache
++ XDG_CACHE_HOME=/tmp/test-flatpak-JBDUYH/home/cache
++ export XDG_CONFIG_HOME=/tmp/test-flatpak-JBDUYH/home/config
++ XDG_CONFIG_HOME=/tmp/test-flatpak-JBDUYH/home/config
++ export XDG_DATA_HOME=/tmp/test-flatpak-JBDUYH/home/share
++ XDG_DATA_HOME=/tmp/test-flatpak-JBDUYH/home/share
++ export XDG_RUNTIME_DIR=/tmp/test-flatpak-JBDUYH/runtime
++ XDG_RUNTIME_DIR=/tmp/test-flatpak-JBDUYH/runtime
++ export USERDIR=/tmp/test-flatpak-JBDUYH/home/share/flatpak
++ USERDIR=/tmp/test-flatpak-JBDUYH/home/share/flatpak
++ export SYSTEMDIR=/tmp/test-flatpak-JBDUYH/system
++ SYSTEMDIR=/tmp/test-flatpak-JBDUYH/system
+++ flatpak --default-arch
++ export ARCH=x86_64
++ ARCH=x86_64
++ '[' x == xyes ']'
++ export FL_DIR=/tmp/test-flatpak-JBDUYH/home/share/flatpak
++ FL_DIR=/tmp/test-flatpak-JBDUYH/home/share/flatpak
++ export U=--user
++ U=--user
++ export INVERT_U=--system
++ INVERT_U=--system
++ '[' x == xyes ']'
++ export 'FLATPAK= flatpak'
++ FLATPAK=' flatpak'
++ export FL_GPG_HOMEDIR=/tmp/test-flatpak-JBDUYH/gpghome
++ FL_GPG_HOMEDIR=/tmp/test-flatpak-JBDUYH/gpghome
++ export FL_GPG_HOMEDIR2=/tmp/test-flatpak-JBDUYH/gpghome2
++ FL_GPG_HOMEDIR2=/tmp/test-flatpak-JBDUYH/gpghome2
++ mkdir -p /tmp/test-flatpak-JBDUYH/gpghome
++ mkdir -p /tmp/test-flatpak-JBDUYH/gpghome2
+++ dirname /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-extensions.sh
++ cp /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-keyring/pubring.gpg /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-keyring/secring.gpg /tmp/test-flatpak-JBDUYH/gpghome/
+++ dirname /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-extensions.sh
++ cp /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-keyring2/pubring.gpg /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-keyring2/secring.gpg /tmp/test-flatpak-JBDUYH/gpghome2/
++ export FL_GPG_ID=7B0961FD
++ FL_GPG_ID=7B0961FD
++ export FL_GPG_ID2=B2314EFC
++ FL_GPG_ID2=B2314EFC
++ export 'FL_GPGARGS=--gpg-homedir=/tmp/test-flatpak-JBDUYH/gpghome --gpg-sign=7B0961FD'
++ FL_GPGARGS='--gpg-homedir=/tmp/test-flatpak-JBDUYH/gpghome --gpg-sign=7B0961FD'
++ export 'FL_GPGARGS2=--gpg-homedir=/tmp/test-flatpak-JBDUYH/gpghome2 --gpg-sign=B2314EFC'
++ FL_GPGARGS2='--gpg-homedir=/tmp/test-flatpak-JBDUYH/gpghome2 --gpg-sign=B2314EFC'
++ export FL_GPG_BASE64=mQENBFbPBvoBCADWbz5O+XzuyN+dDExK81pci+gIzBNWB+7SsN0EgoJppKgwBCX+Bd6ERe9Yz0nJbJB/tjazRp7MnnoPnh6fXnhIbHA766/Eciy4sL5X8laqDmWmROCqCe79QZH/w6vYTKsDmoLQrw9eKRP1ilCvECNGcVdhIyfTDlNrU//uy5U4h2PVUz1/Al87lvaJnrj5423m5GnX+qpEG8mmpmcw52lvXNPuC95ykylPQJjI0WnOuaTcxzRhm5eHPkqKQ+nPIS+66iw1SFdobYuye/vg/rDiyp8uyQkh7FWXnzHxz4J8ovesnrCM7pKI4VEHCnZ4/sj2v9E3l0wJlqZxLTULaV3lABEBAAG0D1hkZy1hcHAgdGVzdGluZ4kBOAQTAQIAIgUCVs8G+gIbAwYLCQgHAwIGFQgCCQoLBBYCAwECHgECF4AACgkQE4sx4HsJYf2DiAf7BQ8anU3CgYpJjuO2rT8jQPO0jGRCNaPyaeAcBx8IjFkjf8daKMPCAt6gQioEpC8OhDig86Bl5piYOB7L7JSB53mgUrADJXhgC/dG4soCt7/U4wW30MseXdlXSOqHGApblF/bIs4B30OBGReBj3DcWIqyb48GraSKlPlaCpkZFySNEAcGUCeCqbbygxCQAM8MDq9FgVRk5oVrE/nAUm6oScEBhseoB7+CaHaRTmLoe/SBs0z2AJ7alIH1Sv4X3mQXpfsAIcWf3Zu2MZydF/Vuh8vTMROwPYtOVEtGxZvEBN3h5uc88dHSk928maqsop9T6oEwM43mBKCOu1gdAOw4OLkBDQRWzwb6AQgAx/XuEaQvdI3J2YYmOE6RY0jJZXLauXH46cJR4q70mlDev/OqYKTSLlo4q06D4ozCwzTYflppDak7kmjWMN224/u1koqFOtF76LsglAeLaQmweWmX0ecbPrzFYaX30kaQAqQ9Wk0PRe0+arRzWDWfUv3qX3y1decKUrBCuEC6WvVVwooWs+zX0cUBS8CROhazTjvXFAz36mhK0u+B3WCBlK+T2tIPOjLjlYgzYARw+X7/J6B3C798r2Hw/yXqCDcKLrq7WWUB33kv3buuG2G6LUamctdD8IsTBxi+nIjAvQITFqq4cPbbXAJGaAnWGuLOddQ9e/GhCOI4JjopRnnjOwARAQABiQEfBBgBAgAJBQJWzwb6AhsMAAoJEBOLMeB7CWH9TC8H/A6oreCxeiL8DPOWN29OaQ5sEw7Dg7bnLSZLu8aREgwfCiFSv0numOABjn/G89Y5M6NiEXFZZhUa+SXOALiBLUy98O84lyp9hlP9qGbWRgBXwe5vOAJERqtoUwR5bygpAw5Nc4y3wddPC4vH7upJ8ftU/eEFtPdI0cKrrAZDFdhXFp3RxdCC6fD62wbofE0mo1Ea1iD3xqVh2t7jfWN1RhMV308htHRGkkmWcEbbvHqugwL6dWZEvQmLYi6/7tQyA1KdG4AZksBP/MBi3t2hthRqQx1v52JwdCaZNuItuEe5rWXhfvoGxPoqYZt9ZPjna6yJfcfJwPbMfjNwX2LR4p4=
++ FL_GPG_BASE64=mQENBFbPBvoBCADWbz5O+XzuyN+dDExK81pci+gIzBNWB+7SsN0EgoJppKgwBCX+Bd6ERe9Yz0nJbJB/tjazRp7MnnoPnh6fXnhIbHA766/Eciy4sL5X8laqDmWmROCqCe79QZH/w6vYTKsDmoLQrw9eKRP1ilCvECNGcVdhIyfTDlNrU//uy5U4h2PVUz1/Al87lvaJnrj5423m5GnX+qpEG8mmpmcw52lvXNPuC95ykylPQJjI0WnOuaTcxzRhm5eHPkqKQ+nPIS+66iw1SFdobYuye/vg/rDiyp8uyQkh7FWXnzHxz4J8ovesnrCM7pKI4VEHCnZ4/sj2v9E3l0wJlqZxLTULaV3lABEBAAG0D1hkZy1hcHAgdGVzdGluZ4kBOAQTAQIAIgUCVs8G+gIbAwYLCQgHAwIGFQgCCQoLBBYCAwECHgECF4AACgkQE4sx4HsJYf2DiAf7BQ8anU3CgYpJjuO2rT8jQPO0jGRCNaPyaeAcBx8IjFkjf8daKMPCAt6gQioEpC8OhDig86Bl5piYOB7L7JSB53mgUrADJXhgC/dG4soCt7/U4wW30MseXdlXSOqHGApblF/bIs4B30OBGReBj3DcWIqyb48GraSKlPlaCpkZFySNEAcGUCeCqbbygxCQAM8MDq9FgVRk5oVrE/nAUm6oScEBhseoB7+CaHaRTmLoe/SBs0z2AJ7alIH1Sv4X3mQXpfsAIcWf3Zu2MZydF/Vuh8vTMROwPYtOVEtGxZvEBN3h5uc88dHSk928maqsop9T6oEwM43mBKCOu1gdAOw4OLkBDQRWzwb6AQgAx/XuEaQvdI3J2YYmOE6RY0jJZXLauXH46cJR4q70mlDev/OqYKTSLlo4q06D4ozCwzTYflppDak7kmjWMN224/u1koqFOtF76LsglAeLaQmweWmX0ecbPrzFYaX30kaQAqQ9Wk0PRe0+arRzWDWfUv3qX3y1decKUrBCuEC6WvVVwooWs+zX0cUBS8CROhazTjvXFAz36mhK0u+B3WCBlK+T2tIPOjLjlYgzYARw+X7/J6B3C798r2Hw/yXqCDcKLrq7WWUB33kv3buuG2G6LUamctdD8IsTBxi+nIjAvQITFqq4cPbbXAJGaAnWGuLOddQ9e/GhCOI4JjopRnnjOwARAQABiQEfBBgBAgAJBQJWzwb6AhsMAAoJEBOLMeB7CWH9TC8H/A6oreCxeiL8DPOWN29OaQ5sEw7Dg7bnLSZLu8aREgwfCiFSv0numOABjn/G89Y5M6NiEXFZZhUa+SXOALiBLUy98O84lyp9hlP9qGbWRgBXwe5vOAJERqtoUwR5bygpAw5Nc4y3wddPC4vH7upJ8ftU/eEFtPdI0cKrrAZDFdhXFp3RxdCC6fD62wbofE0mo1Ea1iD3xqVh2t7jfWN1RhMV308htHRGkkmWcEbbvHqugwL6dWZEvQmLYi6/7tQyA1KdG4AZksBP/MBi3t2hthRqQx1v52JwdCaZNuItuEe5rWXhfvoGxPoqYZt9ZPjna6yJfcfJwPbMfjNwX2LR4p4=
++ export FL_GPG_BASE642=mQENBFkSyx4BCACq/8XFcF+NTpJKfoo8F6YyR8RQXww6kCV47zN78Dt7aCh43WSYLRUBRt1tW5MRT8R60pwCsGvKnFiNS2Vqe4T1IW4mDnFMZIZJXdNVwKUqVBPL/jzkIDnQ9NXtuPNH0qET6VhYnb9aykLo/MiBmx6q+4MvYd/qwiN8kstRifRIxjjZx6wsg+muY6yx9fZKxlgvhc3nsrl3oyDo7/+V+b3heYLtMCQFwlHRKz3Yf2X9H0aUSbDYcgTy6w3q94HVNCpJSqeiR+kBG175BQYKR2l7WYdaVPFf5LMEvAJh0SGnqu77X+8TYYRQiiBB5fYjGOeHfOh6uH5GAJRQymVIJwy/ABEBAAG0KkZsYXRwYWsgKFRlc3Qga2V5IDIpIDxmbGF0cGFrQGZsYXRwYWsub3JnPokBOAQTAQIAIgUCWRLLHgIbAwYLCQgHAwIGFQgCCQoLBBYCAwECHgECF4AACgkQdZ9f0LIxTvyeUQf/euAZpipXBkGWxeW4G10r1QRi2tZAWNeLpy8SB17eo9E6yB61SdH80jALborVs/plnZzKcFf+nLvjCn51FLMh6QPL3S+079WHsed//qtUWfbJ85hLevfCMTZMLktUmqwwUh238WW/gKtbUjYOqr1IZSMBoMiQtc0iOVBP7HUdhYigxTKvs/MBEGHANeQkY07ZnX9oFXElOo+EIPAHScwEOSwEVrXUVHpQODzIfjOoPUHWAZtM1yJT+iWmVHe4HtU8CyBnPyUcnTmTWKr92QmgfWkb1T7ugT5gXt/6ZlYAaZGnr9yNuSk3MMhDMOyldtJBM5Zl8eScE9KBf7pRJoxnMLkBDQRZEsseAQgAvA29IyiJpB+jUHj3MOyVyTBOmvLme+0Ndhpt/mTh+swchJUvzb0IzQS9Le5yVAvn+ppAtDCMb+bV4Xh5zrbiH0Hu0qwK4Qk+KcIKRE8ImDiUM8NFE2SZoomZSsgZ1NBWbAdEyVpkBfrt3Dd8FssMrwPF6kqo02TZr7Pxng+BEHUZT6jPCxueqyXyv2cLbQMe1H0U7klsxPmnnIYUqdwOmPxUspVEYP9oJb5y123mx0yj5JuYdZMjWbP3cRLox1RKIlFWgQqOn2yJiEoWzpqdbtb7sE3ggnbZKJED0ZxUZIakjnyMhX+GAEA8ZMZ6+HfDt1iHV8qHcYiLW5A3AQTxZwARAQABiQEfBBgBAgAJBQJZEsseAhsMAAoJEHWfX9CyMU78Ns4IAJRQ5UJ9KkeZClHm1EjYlgsAq1UJr9wgbyBFKTEkGZ/CAvVmgg+BUXcN/SPAkELbEAOJZTyv8C5cuJC49iFHOxUbRZXZ5eN2SvhZzl+5gep2uHwVLdqRIxFDTHbLWnmtHxPeU7IRA9u86q3wV1N0pD7kreNN7BWKY3/tI33hY2/XVVFy0MN5sutPn+lVK66MqAHqtode5xqqz9Z8LmS7LlqokQkAytcGd6Xqsx99NTk8kk3bnk9HWsAvDO8tRZroeseKeRNmbhGvCNUxPSB6bpYBJLvQtjA9ZVv6sNm0E+SuiXKizZkBGO5AH50pDoy0+MCGoOhwwXeY5+1kZAOzkMI=
++ FL_GPG_BASE642=mQENBFkSyx4BCACq/8XFcF+NTpJKfoo8F6YyR8RQXww6kCV47zN78Dt7aCh43WSYLRUBRt1tW5MRT8R60pwCsGvKnFiNS2Vqe4T1IW4mDnFMZIZJXdNVwKUqVBPL/jzkIDnQ9NXtuPNH0qET6VhYnb9aykLo/MiBmx6q+4MvYd/qwiN8kstRifRIxjjZx6wsg+muY6yx9fZKxlgvhc3nsrl3oyDo7/+V+b3heYLtMCQFwlHRKz3Yf2X9H0aUSbDYcgTy6w3q94HVNCpJSqeiR+kBG175BQYKR2l7WYdaVPFf5LMEvAJh0SGnqu77X+8TYYRQiiBB5fYjGOeHfOh6uH5GAJRQymVIJwy/ABEBAAG0KkZsYXRwYWsgKFRlc3Qga2V5IDIpIDxmbGF0cGFrQGZsYXRwYWsub3JnPokBOAQTAQIAIgUCWRLLHgIbAwYLCQgHAwIGFQgCCQoLBBYCAwECHgECF4AACgkQdZ9f0LIxTvyeUQf/euAZpipXBkGWxeW4G10r1QRi2tZAWNeLpy8SB17eo9E6yB61SdH80jALborVs/plnZzKcFf+nLvjCn51FLMh6QPL3S+079WHsed//qtUWfbJ85hLevfCMTZMLktUmqwwUh238WW/gKtbUjYOqr1IZSMBoMiQtc0iOVBP7HUdhYigxTKvs/MBEGHANeQkY07ZnX9oFXElOo+EIPAHScwEOSwEVrXUVHpQODzIfjOoPUHWAZtM1yJT+iWmVHe4HtU8CyBnPyUcnTmTWKr92QmgfWkb1T7ugT5gXt/6ZlYAaZGnr9yNuSk3MMhDMOyldtJBM5Zl8eScE9KBf7pRJoxnMLkBDQRZEsseAQgAvA29IyiJpB+jUHj3MOyVyTBOmvLme+0Ndhpt/mTh+swchJUvzb0IzQS9Le5yVAvn+ppAtDCMb+bV4Xh5zrbiH0Hu0qwK4Qk+KcIKRE8ImDiUM8NFE2SZoomZSsgZ1NBWbAdEyVpkBfrt3Dd8FssMrwPF6kqo02TZr7Pxng+BEHUZT6jPCxueqyXyv2cLbQMe1H0U7klsxPmnnIYUqdwOmPxUspVEYP9oJb5y123mx0yj5JuYdZMjWbP3cRLox1RKIlFWgQqOn2yJiEoWzpqdbtb7sE3ggnbZKJED0ZxUZIakjnyMhX+GAEA8ZMZ6+HfDt1iHV8qHcYiLW5A3AQTxZwARAQABiQEfBBgBAgAJBQJZEsseAhsMAAoJEHWfX9CyMU78Ns4IAJRQ5UJ9KkeZClHm1EjYlgsAq1UJr9wgbyBFKTEkGZ/CAvVmgg+BUXcN/SPAkELbEAOJZTyv8C5cuJC49iFHOxUbRZXZ5eN2SvhZzl+5gep2uHwVLdqRIxFDTHbLWnmtHxPeU7IRA9u86q3wV1N0pD7kreNN7BWKY3/tI33hY2/XVVFy0MN5sutPn+lVK66MqAHqtode5xqqz9Z8LmS7LlqokQkAytcGd6Xqsx99NTk8kk3bnk9HWsAvDO8tRZroeseKeRNmbhGvCNUxPSB6bpYBJLvQtjA9ZVv6sNm0E+SuiXKizZkBGO5AH50pDoy0+MCGoOhwwXeY5+1kZAOzkMI=
++ _flatpak_bwrap_works=
++ '[' -z '' ']'
++ _flatpak_bwrap_works=true
++ sed s#@testdir@#/nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak# /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/session.conf.in
++ dbus-daemon --fork --config-file=session.conf --print-address=3 --print-pid=4
+++ cat dbus-session-bus-address
++ export DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-vGvuQOwqHW,guid=5450f0fcc69773b3e08235935d51d619
++ DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-vGvuQOwqHW,guid=5450f0fcc69773b3e08235935d51d619
+++ cat dbus-session-bus-pid
++ DBUS_SESSION_BUS_PID=899
++ /nix/store/k8lhqzpaaymshchz8ky3z4653h4kln9d-coreutils-8.31/bin/kill -0 899
++ trap cleanup EXIT
+ skip_without_bwrap
+ true
+ return 0
+ echo 1..2
1..2
+ mkdir -p repos
+ ostree init --repo=repos/test --mode=archive-z2
++ dirname /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/test-extensions.sh
+ . /nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/libexec/installed-tests/flatpak/make-test-runtime.sh repos/test org.test.Platform master '' bash ls cat echo readlink
++ set -e
++ set +x
ln: failed to create symbolic link '/tmp/tmp.y1ToYiWp1c/usr/bin/bash': File exists
gpg-connect-agent: no running gpg-agent - starting '/nix/store/j026d9911bzawfwr8w71bj471h452f47-gnupg-2.2.17/bin/gpg-agent'
gpg-connect-agent: waiting for the agent to come up ... (5s)
gpg-connect-agent: connection to agent established
OK closing connection
fusermount: failed to unmount /tmp/test-flatpak-JBDUYH/runtime/doc: No such file or directory
FAIL: flatpak/test-extensions.sh.test (Child process exited with code 1)
(12.05 seconds)
error: command `env FLATPAK_TESTS_DEBUG=1 gnome-desktop-testing-runner -d '/nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/share' --timeout 3600 flatpak/test-extensions.sh.test 2>&1' did not succeed (exit code 2)
(31.45 seconds)
command `env FLATPAK_TESTS_DEBUG=1 gnome-desktop-testing-runner -d '/nix/store/dbk0crsrbwi5l75hsnc2047bq94q0x0m-flatpak-1.4.2-installedTests/share' --timeout 3600 flatpak/test-extensions.sh.test 2>&1' did not succeed (exit code 2)

```

worldofpeace commented:
I don't think I know where to start to fix that test @jtojnar. Similar quoting issues?

Noting that most of the cases pass I think we should merge this.
What do you think?

jtojnar commented:
I usually remove unsetting of `-x` mode  from the test scripts and then change the test command as above.

If you do not feel like trying to decipher the shell scripts, merging is okay with me and I will take a look when I get back.

worldofpeace commented:
> If you do not feel like trying to decipher the shell scripts, merging is okay with me, and I will take a look when I get back.

Thanks :+1:
